lighthouse-cli/cli-flags.js

```diff
@@ -12,19 +12,32 @@ const pkg = require('../package.json');
 const printer = require('./printer');
 
 /**
- * @param {string=} manualArgv
+ * @param {(string | string[])[]} arr
+ * @return {string[]}
+ */
+function flatten(arr) {
+  /** @type {string[]} */
+  const result = [];
+  return result.concat(...arr);
+}
+
+/**
+ * @param {string[]} manualArgv
  * @return {LH.CliFlags}
  */
-function getFlags(manualArgv) {
-  // @ts-ignore yargs() is incorrectly typed as not returning itself
-  const y = manualArgv ? yargs(manualArgv) : yargs;
-  return y.help('help')
+function getFlags(...manualArgv) {
+  const y = manualArgv.length ? yargs(manualArgv) : yargs;
```

Comment:
this ends up somewhat confusing from the caller's perspective since it's no longer explicitly optional (even though we usually call it with no args) and when we do pass an argument it's still a single string (it's annoying that the `yargs` types require this because [`yargs` definitely supports parsing just a string](https://github.com/yargs/yargs-parser/blob/981e1512d15a0af757a74a9cd75c448259aa8efa/index.js#L9-L11)).

What about keeping it `string=` and doing `const y = manualArgv ? yargs([manualArgv]) : yargs;` instead?

Reply:
I'm confused on how it's confusing to the caller?

if reverting the change, would you want the code using multiple args (admittedly it's just test code) to look like this?

```
getFlags([
	'http://www.example.com',
	'--only-categories=performance,seo',
	'--skipAudits=unused-javascript,redirects',
	'--skipAudits=bootup-time',
].join(' '));
```

or this?
```
getFlags('http://www.example.com --only-categories=performance,seo --skipAudits=unused-javascript,redirects --skipAudits=bootup-time');
```

Reply:
> if reverting the change, would you want the code using multiple args (admittedly it's just test code) to look like this?

yes from me. Maybe it's just that I haven't been completely indoctrinated to rest parameters yet, but to me an optional `manualArgv?: string` parameter is a lot clearer than `...manaulArgv: string[]` at informing that the argument is optional. It's not a huge deal since it's not a commonly used interface, but it's only test code that uses this, so it makes sense to me to make the real use case clear in the signature.